CONTRIBUTING.md

```diff
@@ -151,3 +151,22 @@ Here are some links to help guide you on style as recommended by Dart:
 * Design: <https://dart.dev/guides/language/effective-dart/design>
 
 We recommend following these same guidelines for any of your own packages you may create for the ecosystem, as well.
+
+## Adding a New Component
+
+1. Create the dart code for constructing the module in lib/src
+
+1. Create testing code in test/
+
+1. Create a generator in gen
+
+   1. Add generator to gen/generate.dart
+
+   1. Run dart gen/generate.dart to make sure generation works.
+
+1. Add documentation to doc/name.dart
+
+1. How to test documentation in a dart container
```

Comment:
the dev container already has these installed it seems?

Reply:
I don't follow.  These are instructions for the component developer to go through in adding a component to ROHD-HCL.  Only the last bullet asks the question: how do you see the output of documentation (including schematics) -- I want to document how to check this out before deployment.  If it means using the dev container, I think we need step-by-step instructions on how to use to check your documentation.

Reply:
The file `tool/gh_codespaces/run_setup.sh` does necessary setup including installing open cad suite and d3.  This is the script used in `.devcontainer/devcontainer.json` which is used by VS Code to build a container for ROHD-HCL development, and also is used by GitHub codespaces.  A pointer to how to use the container and/or saying to run the requirements in that script (if needed) might make more sense?

Reply:
I think that may help with the last line of checking documentation / schematics.  It would be good to have a recipe for seeing the output so people can tweak their documentation / schematic examples locally.

And to add a new Widget, we need to tell people that in addition to adding lib/src/Widget.dart and test/Widget_test.dart, they need to also add gen/Widget_gen.dart to create an example for schematic, then modify gen/generate.dart to call that Widget generator, then reference the schematic in doc/components/Widget.md.  Finally, we need a pointer how to invoke via vscode and/or codespaces to check the schematic output (in my case the output was empty because of a naming bug).

I did file a bug that if you do all this and put a link to the final site in your pull request, it will fail because the link is dead and the checking is strict on that.

Reply:
Ok, I think once the Config generation stuff is merged in this will be more streamlined, but I agree some instructions here are helpful.

I'm not sure the best way to handle that issue.  Perhaps the right way is to add links after the main code has merged in?